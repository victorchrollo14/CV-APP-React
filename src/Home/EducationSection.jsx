import { EduContext } from "../App";
import { AddButton, DeleteButton } from "./Button";
import { Heading } from "./Heading";
import { Input } from "./Input";
import { useContext, useState } from "react";

let initialObj = {
  degree: "",
  from: "",
  to: "",
  subject: "",
  university: "",
};

export function EduSection() {
  return (
    <section className="form-sections edu-section">
      <Heading>Education</Heading>
      <EduList />
    </section>
  );
}

function EduList() {
  const { eduList, setEduList } = useContext(EduContext);
  const [tempObj, setTempObj] = useState(initialObj);

  const handleDegChange = (e) => {
    setTempObj({ ...tempObj, degree: e.target.value });
  };

  const handleFromChange = (e) => {
    setTempObj({ ...tempObj, from: e.target.value });
  };

  const handleToChange = (e) => {
    setTempObj({ ...tempObj, to: e.target.value });
  };

  const handleSubChange = (e) => {
    setTempObj({ ...tempObj, subject: e.target.value });
  };

  const handleUniChange = (e) => {
    setTempObj({ ...tempObj, university: e.target.value });
  };

  const addItem = () => {
    const { degree, from, to, subject, university } = tempObj;
    setEduList([
      ...eduList,
      {
        id: crypto.randomUUID(),
        degree: degree,
        from: from,
        to: to,
        subject: subject,
        university: university,
      },
    ]);
    setTempObj(initialObj);
  };

  const delItem = (e, id) => {
      setEduList((currentlist) => currentlist.filter(ed => ed.id !== id))
  }

  const list = eduList.map(({ id, degree, from, to, subject, university }) => {
    if (eduList) {
      return (
        <li key={id} >
          <Input name={degree} read={true}>
            Degree{" "}
          </Input>
          <div className="same-row">
            <Input name={from} read={true}>
              From
            </Input>
            <Input name={to} read={true}>
              To
            </Input>
          </div>
          <Input name={subject} read={true}>
            Subject Name
          </Input>
          <Input name={university} read={true}>
            University/School Name
          </Input>
          <div className="same-row">
            <DeleteButton onClick={(e) => delItem(e, id)} />
          </div>
        </li>
      );
    }
    return null;
  });

  return (
    <ul className="list gap-btw">
      <li>
        <Input name={tempObj.degree} onChange={handleDegChange}>
          Degree
        </Input>
        <div className="same-row">
          <Input name={tempObj.from} onChange={handleFromChange}>
            From
          </Input>
          <Input name={tempObj.to} onChange={handleToChange}>
            To
          </Input>
        </div>
        <Input name={tempObj.subject} onChange={handleSubChange}>
          Subject Name
        </Input>
        <Input name={tempObj.university} onChange={handleUniChange}>
          University/School Name
        </Input>
        <AddButton onClick={addItem} />
      </li>
      {list}
    </ul>
  );
}
