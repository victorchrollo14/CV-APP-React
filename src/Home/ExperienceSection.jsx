import { useCallback, useContext, useState } from "react";
import { AddButton, DeleteButton } from "./Button";
import { Heading } from "./Heading";
import { Input } from "./Input";
import { InitExpForm } from "./InitExpForm";
import { WorkContext } from "../App";

export function ExpSection() {
  return (
    <section className="exp-section form-sections">
      <Heading>Experience</Heading>
      <ExpList />
    </section>
  );
}

function ExpList() {
  const { workList, setWorkList } = useContext(WorkContext);

  const delItem = (id) => {
    setWorkList((currentlist) => currentlist.filter((work) => work.id !== id));
  };

  const itemsList = workList.map(({ id, company, from, to, city, role }) => {
    return (
      <li key={id}>
        <Input name={company}>Company </Input>
        <div className="same-row">
          <Input name={from}>From</Input>
          <Input name={to}> to</Input>
        </div>
        <Input name={city}>City </Input>
        <Input name={role}>Role</Input>
        <DeleteButton onClick={() => delItem(id)} />
      </li>
    );
  });

  return (
    <ul className="list">
      <InitExpForm />
      {itemsList}
    </ul>
  );
}
