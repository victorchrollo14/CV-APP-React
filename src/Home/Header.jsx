import { Link } from "react-router-dom";

export function Header() {
  return (
    <>
      <header className="header" id="header">
        <h1>CV Builder</h1>
        <a href="#Resume-preview" className="preview-btn">
          {" "}
          Preview
        </a>
      </header>
    </>
  );
}
