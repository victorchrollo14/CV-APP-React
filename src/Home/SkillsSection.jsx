import { SkillContext } from "../App";
import { Heading } from "./Heading";
import React, { useContext, useState } from "react";

export function SkillsSection() {
  const [newInput, setNewInput] = useState("");
  const { skillList, setSkillList } = useContext(SkillContext);

  const addInput = () => {
    if (newInput !== "") {
      setSkillList([...skillList, { id: crypto.randomUUID(), name: newInput }]);
      setNewInput("");
    }
  };

  const deleteInput = (id) => {
    setSkillList((currentlist) =>
      currentlist.filter((skill) => {
        if (skill.id !== id) {
          return skill;
        }
      })
    );
  };

  return (
    <section className="form-sections edu-section">
      <Heading>Skills</Heading>

      <div>
        <input
          placeholder="Enter a new skill"
          type="text"
          className="input-button margin"
          value={newInput}
          onChange={(e) => setNewInput(e.target.value)}
        />
        <button className={"btn add-btn"} onClick={addInput}>
          Add
        </button>

        {skillList.map((skill) => (
          <div key={skill.id}>
            <input
              type="text"
              value={skill.name}
              className="input-button margin"
              style={{ color: "white" }}
              readOnly
            />
            <button
              className={"btn del-btn"}
              onClick={() => deleteInput(skill.id)}
            >
              Delete
            </button>
          </div>
        ))}
      </div>
    </section>
  );
}
