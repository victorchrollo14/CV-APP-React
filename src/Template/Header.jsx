import { Link } from "react-router-dom"

export function Header() {
  return (
    <header className="header">
      <h1>Template Preview</h1>
      <button className="preview-btn">
        <Link to="/" className="links">Go Back</Link>
      </button>
    </header>
  );
}


