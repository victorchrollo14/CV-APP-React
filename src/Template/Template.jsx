function Template() {
  return (
    <>
      <div className="template">
        <h1>Welcome to template</h1>
      </div>
    </>
  );
}

export default Template;
