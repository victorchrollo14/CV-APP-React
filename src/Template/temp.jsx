import React, { useState } from "react";

export function Temp() {
  const [heading, setHeading] = useState("Magnificent Monkeys");

  const clickHandler = () => {
    setHeading("Radical Rhinos");
  };

  return (
    <>
      <button type="button" onClick={clickHandler}>
        Click Me
      </button>
      <h1>{heading}</h1>
    </>
  );
}
