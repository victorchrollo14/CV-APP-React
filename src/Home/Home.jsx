import "./style.css"

import { Header } from './Header'
import { Form } from './Form'
import { DownLoadResume } from "../Template/Resume"

function Home() {

  return (
    <>
      <Header />
      <Form />
      <DownLoadResume />
    </>
  )
}

export default Home
