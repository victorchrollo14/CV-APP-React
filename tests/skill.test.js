import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { Header } from "../src/Home/Header";
import { Temp } from "../src/Template/temp";
import { SkillsSection } from "../src/Home/SkillsSection";

describe("App component", () => {
  it("renders correct heading", () => {
    render(<Header />);
    expect(screen.getByRole("heading", { name: "CV Builder" }));
  });
});

describe("App component", () => {
  it("renders magnificant monkeys", () => {
    expect(<Temp />).toMatchSnapshot();
  });

  it("radical rhinos after button click", async () => {
    const user = userEvent.setup();
    render(<Temp />);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  });
});

describe("App component", () => {
    it("Renders correct input", () => {
        render(<SkillsSection />)
        let input = screen.getByPlaceholderText("Enter a new Skill");
        expect(input.textContent).toMatch("");
    })
})
