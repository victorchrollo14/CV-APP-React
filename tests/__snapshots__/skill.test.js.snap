// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders magnificant monkeys 1`] = `<Temp />`;
